proposals/0000-wire-protocol.md

```diff
@@ -0,0 +1,435 @@
+
+Title: **DEP-0000: Hypercore Wire Protocol**
+
+Short Name: `0000-wire-protocol`
+
+Type: Standard
+
+Status: Undefined (as of 2018-02-04)
+
+Github PR: (add HTTPS link here after PR is opened)
+
+Authors: [Paul Frazee](https://github.com/pfrazee), [Bryan Newbold](https://github.com/bnewbold)
+
+
+# Summary
+[summary]: #summary
+
+This DEP describes the Hypercore wire protocol: a transport-agnostic message stream spoken between nodes in a swarm of hypercore network peers (including Dat clients). The wire protocol includes mechanisms for framing, stream encryption, and feed key authentication.
+
+
+# Motivation
+[motivation]: #motivation
+
+The protocol described here is already in use as of 2017 (by Hypercore, Dat, and Beaker Browser users), and was partially described in an earlier [whitepaper][whitepaper]. This document fills in some additional details.
+
+[whitepaper]: https://github.com/datprotocol/whitepaper
+
+
+# Stream Connections
+[stream-details]: #stream-details
+
+The Dat wire protocol depends on the use of a binary transport protocol which provides the following semantics:
+
+- reliable delivery (no dropped or partial messages)
+- in-order delivery of messages
+
+Peers wishing to connect need to discover each other using some mechanism or another (see forthcoming DEPs on some options; this process is modular and swappable), and both need to have the public key for the primary hypercore they wish to exchange.
+
+Messages are framed by the Dat protocol itself (see Messages section for details).
+
+
+## Channels
+[channels]: #channels
+
+Multiple hypercore feeds can be synchronized over the same protocol connection. Messages pertaining to the separate channels are tagged with an id for disambiguation.
+
+Note that at least one feed is necessary for each connection (for handshaking to succeed), and that the first feed is the one used for discovery and as an encryption key.
+
+To initiate a new channel (after the primary is established), TODO
```

Comment:
... send an (encrypted) Feed message followed by an Info.

It's a little unclear which party *should* initiate this (I supposed either can at any time), and what to do if both parties send a Feed message with the same index and different feed public keys (aka, how is the dispute resolved?).

Reply:
I'll expand on that